src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""A writer for wfx format files."""
+
+import os.path
+import math
+from cclib.parser import utils
+
+from . import filewriter
+
+class WFXWriter(filewriter.Writer):
+    """A writer for wfx files."""
+
+    required_attrs = ['atomcoords', 'atomnos', 'gbasis', 'charge', 'homos', 'mult']
+    keyword = 'GTO'
+    num_orb = {'s':1, 'p':3, 'd':6, 'f':10, 'g':15, 'h':21,
+               'S':1, 'P':3, 'D':6, 'F':10, 'G':15, 'H':21}
+    count_orb = {'s':1, 'p':2, 'd':5, 'f':11, 'g':21, 'h':36,
+                 'S':1, 'P':2, 'D':5, 'F':11, 'G':21, 'H':36}
+
+    @staticmethod
+    def _section(section_name, section_data):
+        """Add opening/closing section_name tags to data."""
+        opening_tag = ['<' + section_name + '>']
+        closing_tag = ['</' + section_name + '>']
+        
+        if isinstance(section_data, list):
+            return opening_tag + section_data + closing_tag
+        elif isinstance(section_data, str):
+            return opening_tag + section_data.split('\n') + closing_tag
+        elif isinstance(section_data, int) or isinstance(section_data, float):
+            return opening_tag + [str(section_data)] + closing_tag
+
```

Comment:
So far its these three only; any un-handled case would throw an exception `Failed to write section`.